plugins/pixart-rf/fu-pxi-device.c

```diff
@@ -599,10 +615,25 @@ static gboolean
 fu_pxi_device_setup (FuDevice *device, GError **error)
 {
 	FuPxiDevice *self = FU_PXI_DEVICE (device);
+	struct hidraw_devinfo hid_raw_info = { 0x0 };
+	g_autofree gchar *devid = NULL;
+	g_autoptr(GString) ble_name = g_string_new (fu_device_get_name (device));
+
+	if (!fu_pxi_device_get_raw_info (self, &hid_raw_info ,error))
+		return FALSE;
 	if (!fu_pxi_device_fw_ota_init (self, error))
 		return FALSE;
 	if (!fu_pxi_device_fw_get_info (self, error))
 		return FALSE;
+
+	/* extra GUID */
+	g_string_ascii_up (ble_name);
+	fu_common_string_replace (ble_name, " ", "_");
```

Comment:
is there any other filtering that might be needed for this string?

Reply:
Do we have any ideas for this? Quite a few plugins do this now. I think the baseline ought to be ASCII, no spaces, uppercase.

Reply:
yeah just with the all the fuzzing stuff you've done lately in mind, it's definitely "untrusted" data.  I think that's spot on, ASCII only, no spaces.  uppercase we need to double check all the other plugins that they're not using lowercase.

Reply:
> we need to double check all the other plugins

At least one is. We can't change the instance IDs now, the GUIDs are essentially _API_.

Reply:
then just got with uppercase for everything new I guess